src/App.js

```diff
@@ -5,27 +5,19 @@ import DopplerIntlProvider from './DopplerIntlProvider';
 import { FormattedMessage } from 'react-intl';
 import Header from './components/Header/Header';
 import Footer from './components/Footer/Footer';
-import axios from 'axios';
-import { HttpDopplerLegacyClient } from './services/doppler-legacy-client';
-import { OnlineSessionManager } from './services/session-manager';
 
 class App extends Component {
-  constructor(props) {
-    super(props);
+  constructor({ locale, dependencies: { sessionManager } }) {
+    super();
 
     this.updateSession = this.updateSession.bind(this);
 
-    this.sessionManager =
-      (props.dependencies && props.dependencies.sessionManager) ||
-      new OnlineSessionManager(
-        (props.dependencies && props.dependencies.dopplerLegacyClient) ||
-          new HttpDopplerLegacyClient(axios, process.env.REACT_APP_API_URL),
-        process.env.REACT_APP_DOPPLER_LEGACY_KEEP_ALIVE_MS,
-      );
+    /** @type { import('./services/session-manager').SessionManager } */
```

Comment:
🚧

Reply:
It looks a little weird but it is the way that I found to allow intellisense for this injected property.